### PR TITLE
Skeleton detachable body parts

### PR DESCRIPTION
**Pull request by flrsh**
[FEATURE] [CLEAN]

## About the PR 

With this PR, skeletons can now detach their limbs and head (without dying!). This PR also slightly cleans up / refactors some surgery code.

**Head**
* Once detached, the skeleton's "eye" is centered on the head object while their controls are still bound to their body
    * This means decapitation does not kill skeletons
    * Skeletons will not talk out of their heads, unfortunately, but rather still their bodies
* Cloths can be used on the head object to clean it
 
**Limbs**
* Once detached, the skeleton's limbs are functionally boomerangs that reattach upon catching and deal more damage than non-skeleton limbs
* "Joint wax" can be applied to empty limb sockets by targeting the respective limb; skeletons receive the "spry" status buff upon application 
    * Skeletons spawn with a small jar of joint wax to incentivize this practice
    * With the spry status buff, users move slightly faster and incur a smaller slowdown when injured 

**To detach:** Traditional surgery *or* crowbar + wrench + crowbar targeting respective body part. 
**To reattach:** Hold part in active hand and target respective body part. 

The sfx are by Virva (though I kinda mangled them), the spry status icon is by Dim, and the joint wax sprite is by Flab! 

https://streamable.com/83ssyk

## Why's this needed? 

I think this is a cool flavor thing that makes skeletons more interesting to play. This also provides a basic framework for some other skeleton things I want to add later. 

## Changelog

```changelog
(u)Flourish
(*)Skeletons can now detach their head and limbs by targeting the respective part and using a wrench, crowbar, and wrench again. Once their head is detached, the skeleton's "eye" remains centered on their head object. Once a limb is detached, it can be used as a boomerang that automatically re-attaches upon catching. 
(*)Skeletons now also spawn with a jar of "joint wax" which can be applied to empty joint sockets for a small status buff. Check out the <a href="https://github.com/goonstation/goonstation/pull/8761" target="_blank">PR page</a> to learn more. Special thanks to Virva, DimWhat, and Flaborized for the sfx and sprites that made this possible!
```

### CODE REVIEW

**Comment by flappybatpal** [comment]
Won't this mean katanas, airlock, firelock suicides and the decap deagle are no longer fatal?

**Comment by Exester509** [comment]
> 

Katana decap will still leave you bleeding a whole lot, and good luck preventing the antag from stealing your head and throwing it into a grinder with a blind, bleeding body. As for the deagle, I believe it removes your head altogether, doesn't it?

**Comment by Confusedxoz** [comment]
Will this make getting item arms easier?

**Comment by flrsh** [comment]
> Won't this mean katanas, airlock, firelock suicides and the decap deagle are no longer fatal to skeletons?

Yes. If that's an issue, I can change it so that only wrench - crowbar - wrench decapitation is nonfatal, and that all other methods are still fatal. Let me know!

> Will this make getting item arms easier?

Only in the sense that there's an additional way to remove them for skeletons.

> In the video, it shows the "eye" of the skeleton initially being on the head but then snapping to the body after some time, whats the deal with that? 

The "eye" snaps to the body when the head is picked up and attached.

**Comment by frawhst** [comment]
the decapitation stuff seems fine; with the katana it's a coup de grace, deagle is an admin weapon, and if someone is trying to surgery your head off as a combat thing then they can just toss your head out an airlock which will create. Issues for you'

**Comment by frawhst** [comment]
tangentially related; could examining the head display whether or not the person is conscious? could be confusing talking to a skeleton head while their body is in crit or otherwise unconscious somewhere and suddenly not getting a response with no indicator why

**Comment by ThatFiveGuys** [comment]
Talking specifically in the video, 15 seconds in without head in hand the camera is centered on the headless body.  Also the text should probably still come out of the head even if its detached, seems funnier when the main body gets dragged away or whatever.

**Comment by flrsh** [comment]
> tangentially related; could examining the head display whether or not the person is conscious? 

Yup, added!

> Talking specifically in the video, 15 seconds in without head in hand the camera is centered on the headless body. 

Ah yeah you're right, that was a bug caused by swapping positions with Mrs. Muggles. Fixed!

>Also the text should probably still come out of the head even if its detached, seems funnier when the main body gets dragged away or whatever. I dont know if the blindness of the disattached head is needed? While I do get that it can be used as a way to disguise as being dead effectively, I think the blindness would make that strat inviable and that strat seems fun.

These are due to code limitations unfortunately! (Unless I wanted to completely refactor how some of these systems work.)

File: _std/macros/ismob.dm [hunk 1]
```diff
@@ -20,6 +20,7 @@
 #define iscow(x) (istype(x, /mob/living/carbon/human) && istype(x:mutantrace, /datum/mutantrace/cow))
 #define ispug(x) (istype(x, /mob/living/carbon/human) && istype(x:mutantrace, /datum/mutantrace/pug))
 #define isfrog(x) (istype(x, /mob/living/carbon/human) && istype(x:mutantrace, /datum/mutantrace/amphibian))
+#define isskeleton(x) (istype(x, /mob/living/carbon/human) && istype(x:mutantrace, /datum/mutantrace/skeleton))
```
Comment (ZeWaka) [inline _std/macros/ismob.dm:23]: rip `IS_BONER` 😩

File: code/procs/jobprocs.dm [hunk 2]
```diff
@@ -681,6 +681,11 @@
 					if (src.limbs.r_arm)
 						qdel(src.limbs.r_arm.remove(0))
 				boutput(src, "<b>Your singular arm makes you feel responsible for crimes you couldn't possibly have committed.</b>" )
+
```
Comment (ZeWaka) [inline code/procs/jobprocs.dm:684]:
```suggestion

	// Special mutantrace items
```

File: code/obj/item/tool/wrench.dm [hunk 3]
```diff
@@ -23,6 +23,10 @@
 		..()
 		BLOCK_SETUP(BLOCK_ROD)
 
+	attack(mob/living/carbon/M as mob, mob/user as mob)
+		if (!wrench_surgery(M, user))
```
Comment (ZeWaka) [inline code/obj/item/tool/wrench.dm:27]: these should really be named like `surgery_wrench` but beyond scope

File: code/datums/mutantraces.dm [hunk 4]
```diff
@@ -1104,18 +1104,49 @@ TYPEINFO(/datum/mutantrace)
...
+						H.visible_message("<span class='notice'>[user] applies some joint wax to [H].</notice>")
+						src.uses--
+						if (!src.uses)
+							src.icon_state = "wax-empty"
+					return
+		..()
+
+	get_desc()
+		if (uses > 0)
+			. += " It looks like it has [uses] applications left."
```
Comment (ZeWaka) [inline code/datums/mutantraces.dm:1149]: what's wrong with it saying 0 left?

File: code/obj/item/organ_holder.dm [hunk 5]
```diff
@@ -767,8 +769,6 @@
 					else
 						return 0
 				var/obj/item/organ/head/newHead = I
-				if (src.donor.client)
-					src.donor.client.mob = new /mob/dead/observer(src.donor)
```
Comment (ZeWaka) [inline code/obj/item/organ_holder.dm:771]: don't see where this got moved

File: code/obj/item/organs/head.dm [hunk 6]
```diff
@@ -333,7 +336,10 @@
 			else
 				boutput(user, "<span class='alert'>[src.name] is already wearing [src.glasses.name]!</span>")
 			return
-
+		if (istype(W, /obj/item/cloth))
+			playsound(src, "sound/items/towel.ogg", 25, 1)
+			user.visible_message("<span class='notice'>[user] [pick("buffs", "shines", "cleans", "wipes", "polishes")] [src] with [W].</span>")
+			return
```
Comment (ZeWaka) [inline code/obj/item/organs/head.dm:342]: don't see how this is cleaning off blood stains and such

File: code/datums/mutantraces.dm [hunk 7]
```diff
@@ -1104,18 +1104,49 @@ TYPEINFO(/datum/mutantrace)
...
 			mob.mob_flags &= ~IS_BONEY
 		. = ..()
 
+/obj/item/joint_wax
+	name = "joint wax"
+	desc = "Does what it says on the jar."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "wax"
+	w_class = W_CLASS_SMALL
+	var/uses = 10
```
Comment (ZeWaka) [inline code/datums/mutantraces.dm:1127]: is there any way planned to refill this

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/procs/jobprocs.dm (hunk 2 above)
Comment (flrsh) [reply code/procs/jobprocs.dm:684]: Such a dishonest way of upping your commit count!

File: code/datums/mutantraces.dm (hunk 4 above)
Comment (flrsh) [reply code/datums/mutantraces.dm:1149]: Fair.

File: code/obj/item/organs/head.dm (hunk 6 above)
Comment (flrsh) [reply code/obj/item/organs/head.dm:342]: It was just for flavor but I can make it clean blood stains and such too.

File: code/datums/mutantraces.dm (hunk 7 above)
Comment (flrsh) [reply code/datums/mutantraces.dm:1127]: Not in this PR but maybe in the future, possibly.

File: code/obj/item/organ_holder.dm (hunk 5 above)
Comment (flrsh) [reply code/obj/item/organ_holder.dm:771]: It's actually unnecessary cuz it only matters for brains, and brains do this already.

File: code/procs/jobprocs.dm (hunk 2 above)
Comment (ZeWaka) [reply code/procs/jobprocs.dm:684]: you don't need to add me as a co-author when merging

File: code/obj/item/organs/head.dm (hunk 6 above)
Comment (ZeWaka) [reply code/obj/item/organs/head.dm]: i thought you meant by
> Cloths can be used on the head object to clean it
that it actually cleaned it lol

**Review by ZeWaka — APPROVED** [review]

File: code/obj/item/mob_parts.dm [hunk 8]
```diff
@@ -259,66 +259,35 @@ ABSTRACT_TYPE(/obj/item/parts)
...
-			if (!(attacher.zone_sel.selecting in list("l_leg","r_leg")))
-				return ..()
-			else if(attachee.limbs.vars["l_leg"] || attachee.limbs.vars["r_leg"])
-				boutput(attacher, "<span class='alert'>[attachee.name] still has one leg!</span>")
-				return
+			attacher.remove_item(src)
 
-			attachee.limbs.l_leg = src
-			attachee.limbs.r_leg = src
+			playsound(attachee, "sound/effects/attach.ogg", 50, 1)
```
Comment (Sovexe) [inline code/obj/item/mob_parts.dm:276]: single quotes are proffered when referencing sound files that don't require string operations

Comment (flrsh) [reply code/obj/item/mob_parts.dm:276]: I always get confused about which to use, thanks!

**Pull request merged**